src/exe/Main.hs

```diff
@@ -99,11 +101,13 @@ application = do
 --   'makeOutputFileName' for the input module to the output directory.
 --   If the output directory does not exist, the output directory and all of
 --   its parent directories are created.
-processInputFile
-  :: Members '[Cancel, Embed IO, GetOpt, Report] r => FilePath -> Sem r ()
+processInputFile :: Members '[Cancel, Embed IO, GetOpt, InputFile, Report] r
+                 => FilePath
+                 -> Sem r ()
 processInputFile inputFilename = do
   input <- embed $ readFile inputFilename
   frontend <- parseFrontend =<< getOpt optFrontend
+  _ <- addInputFile inputFilename input
```

Comment:
Because I remembered a GHC warning I once got for not explicitly assigning such a return value to wildcard. But the GHC probably won't complain in this case, so I will change that.